src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -935,29 +935,130 @@ class C<T> where T : struct
+                // (10,5): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Please use language version 'preview' or greater to use annotations with unconstrained type parameters, or consider adding a 'class', 'struct', or type constraint.
                 //     T? F3;
-                Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "T?").WithLocation(10, 5)
-            );
+                Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "T?").WithArguments("preview").WithLocation(10, 5));
 
             // [NonNullTypes] missing
             comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
             comp.VerifyDiagnostics(
-                // (10,6): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
+                // (10,6): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
                 //     T? F3;
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(10, 6),
-                // (10,5): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.
+                // (10,5): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Please use language version 'preview' or greater to use annotations with unconstrained type parameters, or consider adding a 'class', 'struct', or type constraint.
                 //     T? F3;
-                Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "T?").WithLocation(10, 5)
-            );
+                Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "T?").WithArguments("preview").WithLocation(10, 5));
 
             // https://github.com/dotnet/roslyn/issues/29976: Test with [NonNullTypes(Warnings=false)].
         }
 
-        // https://github.com/dotnet/roslyn/issues/29976: Test `where T : unmanaged`.
+        [Fact]
+        public void GenericType_NoConstraint()
+        {
+            var source =
+@"#nullable enable
+#pragma warning disable 0169
+class C<T>
+{
+    private T F1;
+    private T? F2;
+    private T P1 { get; }
+    private T? P2 { get; }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (5,15): warning CS8618: Non-nullable field 'F1' is uninitialized. Consider declaring the field as nullable.
+                //     private T F1;
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "F1").WithArguments("field", "F1").WithLocation(5, 15),
+                // (6,16): warning CS8618: Non-nullable field 'F2' is uninitialized. Consider declaring the field as nullable.
```

Comment:
>// (6,16): warning CS8618: Non-nullable field 'F2' is uninitialized. Consider declaring the field as nullable. [](start = 16, length = 110)

This looks surprising. Why do we still give this warning?

Reply:
Good catch, thanks.

---
In reply to: [456704718](https://github.com/dotnet/roslyn/pull/45993#discussion_r456704718) [](ancestors = 456704718)